### PR TITLE
feat: create API to return all addresses of a wallet

### PR DESCRIPTION
**Pull request by pedroferreira1**

### CODE REVIEW

File: hathor/wallet/resources/addresses.py [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+            # We have this method only for HD wallet
+            request.setResponseCode(503)
+            return json.dumps({'success': False}, indent=4).encode('utf-8')
+
+        addresses = self.manager.wallet.get_all_addresses()
+
+        data = {
+            'addresses': addresses,
+        }
+        return json.dumps(data, indent=4).encode('utf-8')
```
Comment (obiyankenobi) [inline hathor/wallet/resources/addresses.py:39]: Shouldn't we add `success: True` to maintain our standard?

**Review by obiyankenobi — APPROVED** [review]

File: hathor/wallet/resources/addresses.py (hunk 1 above)
Comment (pedroferreira1) [reply hathor/wallet/resources/addresses.py:39]: Actually I kept the same behaviour of the `AddressResource`, where we return only the address. But I guess it's and old resource, I will add the `success: True`

**Review by jansegre — APPROVED** [review]

File: hathor/wallet/resources/addresses.py [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+
+        if not self.manager.wallet or self.manager.wallet.type != self.manager.wallet.WalletType.HD:
+            # We have this method only for HD wallet
+            request.setResponseCode(503)
+            return json.dumps({'success': False}, indent=4).encode('utf-8')
+
+        addresses = self.manager.wallet.get_all_addresses()
+
+        data = {
+            'addresses': addresses,
```
Comment (msbrogli) [inline hathor/wallet/resources/addresses.py:37]: We must paginate. It feels very unsafe to return all addresses.

**Review by msbrogli — CHANGES_REQUESTED** [review]

**Comment by pedroferreira1** [comment]
This was created at first to help some miners that were using our built in wallet in the full node but it's not needed anymore. They are mining without this API

**Pull request closed without merging**